src/components/ButtonV2/index.tsx

```diff
@@ -0,0 +1,243 @@
+
+    return `
+      background: ${
+        variant === 'neutral'
+          ? theme.colors.neutral.background
+          : theme.colors[variant][backgroundValue]
+      };
+      color: ${
+        variant === 'neutral'
+          ? theme.colors.neutral.text
+          : theme.colors[variant][colorValue]
+      };
+      border: ${
+        variant === 'neutral' || prominence === 'weak'
+          ? `1px solid ${theme.colors[variant].borderWeak}`
+          : 'none'
+      };
+      
+      ${
+        !disabled
+          ? `
+        cursor: pointer;
+        &:hover,:focus-visible {
+          box-shadow: ${theme.shadows.dropdown};
+          outline : 0;
+          background: ${
+            variant === 'neutral'
+              ? theme.colors.neutral.backgroundWeakHover
+              : theme.colors[variant][
+                  `${backgroundValue}Hover` as keyof typeof theme.colors.primary
+                ]
+          };
+          color: ${
+            variant === 'neutral'
+              ? theme.colors.neutral.textHover
+              : theme.colors[variant][
+                  `${colorValue}Hover` as keyof typeof theme.colors.primary
+                ]
+          };
+          border: ${
+            variant === 'neutral' || prominence === 'weak'
+              ? `1px solid ${theme.colors[variant].borderWeakHover}`
+              : 'none'
+          };
+        }
+
+        &:active {
+          outline : 3px solid #4F059940;
+        }
+      `
+          : /* Disabled */ `
+          cursor: not-allowed;
+          color : ${
+            variant === 'neutral'
+              ? theme.colors.neutral.textDisabled
+              : theme.colors[variant][colorValue]
+          };
+          border: ${
+            variant === 'neutral' || prominence === 'weak'
+              ? `1px solid ${
+                  variant === 'neutral'
+                    ? theme.colors.neutral.borderDisabled
+                    : theme.colors[variant].borderWeakDisabled
+                }`
+              : 'none'
+          };
+          `
+      }
+
+      ${
+        extend
+          ? `
+
+          max-width: ${size === 'large' ? '48px' : '32px'};
+          justify-content: ${
+            iconPosition === 'right' ? 'flex-end' : 'flex-start'
+          };
+
+          .button-text { 
+            transition: max-width 0.5s linear;
+            overflow: hidden;
+            max-width: 0;
+            visibility: hidden;
+          }
+
+          &:hover, &:focus {
+
+            max-width: none;
+
+            .button-text {
+              visibility: visible;
+              max-width: 250px;
+            }
+          }
+      `
+          : ''
+      }
+    `
+  }};
+`
+
+const ButtonV2 = ({
+  ariaLabel,
+  disabled = false,
+  className,
+  extend = false,
+  icon,
+  iconPosition = 'left',
+  innerRef,
+  isLoading = false,
+  onClick,
+  prominence = 'strong',
+  size = 'large',
+  text,
+  tooltip,
+  tooltipBaseId,
```

Comment:
I think we could/should have a tooltip almost every time a button is disabled to explain why it is disabled. If we wan't to change ButtonV2 API, let's wait for @matthprost point of view about that (for me it's not a bad thing, it can enforce design system)